app/controllers/RegistrationConfirmationController.scala

```diff
@@ -17,35 +17,40 @@
 package controllers
 
 import controllers.actions._
-import models.SubscriptionID
-
-import javax.inject.Inject
+import pages.SubscriptionIDPage
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import repositories.SessionRepository
 import services.EmailService
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import views.html.RegistrationConfirmationView
 
-import scala.concurrent.ExecutionContext
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
 
-class RegistrationConfirmationController @Inject()(
-                                       override val messagesApi: MessagesApi,
-                                       standardActionSets: StandardActionSets,
-                                       sessionRepository: SessionRepository,
-                                       emailService: EmailService,
-                                       val controllerComponents: MessagesControllerComponents,
-                                       view: RegistrationConfirmationView
-                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+class RegistrationConfirmationController @Inject() (
+  override val messagesApi: MessagesApi,
+  standardActionSets: StandardActionSets,
+  sessionRepository: SessionRepository,
+  emailService: EmailService,
+  val controllerComponents: MessagesControllerComponents,
+  view: RegistrationConfirmationView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
 
   def onPageLoad: Action[AnyContent] = standardActionSets.identifiedWithoutEnrolmentCheck().async {
     implicit request =>
-      val subscriptionId = SubscriptionID("XTCBC0100000001") //Todo: get the subscription ID correctly
-      emailService.sendEmail(request.userAnswers, subscriptionId) flatMap {
-        _ =>
-          sessionRepository.clear(request.userId) map { _ =>
-            Ok(view(subscriptionId.value))
+      request.userAnswers.get(SubscriptionIDPage) match {
+        case Some(subscriptionId) =>
+          emailService.sendEmail(request.userAnswers, subscriptionId) flatMap {
+            _ =>
+              sessionRepository.clear(request.userId) map {
+                _ =>
+                  Ok(view(subscriptionId.value))
+              }
           }
+        case _ => Future.successful(NotImplemented) //TODO Technical difficulties
```

Comment:
@shaileshp0110 Please use Redirect(controllers.routes.ThereIsAProblemController.onPageLoad())
Its in main now

Reply:
@ashwinirudrappa There is ticket related to missing information https://jira.tools.tax.service.gov.uk/browse/DAC6-1861